src/jailer/src/env.rs

```diff
@@ -197,37 +200,63 @@ impl Env {
     }
 
     fn exec_into_new_pid_ns(&mut self, chroot_exec_file: PathBuf) -> Result<()> {
-        // Unshare into a new PID namespace.
-        // The current process will not be moved into the newly created namespace, but its first
-        // child will assume the role of init(1) in the new namespace.
-        // The call is safe because we're invoking a C library function with valid parameters.
-        SyscallReturnCode(unsafe { libc::unshare(libc::CLONE_NEWPID) })
-            .into_empty_result()
-            .map_err(Error::UnshareNewPID)?;
-
         // Compute jailer's total CPU time up to the current time.
         self.jailer_cpu_time_us =
             utils::time::get_time_us(utils::time::ClockType::ProcessCpu) - self.start_time_cpu_us;
 
-        // Duplicate the current process. The child process will belong to the previously created
-        // PID namespace.
-        // TODO: replace the `unshare()` + `fork()` combo with `clone()` if we ever need to
-        //  squeeze every bit of start-up latency we can get
-        let pid = unsafe { libc::fork() };
-        match pid {
-            0 => {
-                // Reset process start time.
-                self.start_time_cpu_us = 0;
-
-                Err(Error::Exec(self.exec_command(chroot_exec_file)))
-            }
-            child_pid => {
-                // Save the PID of the process running the exec file provided
-                // inside <chroot_exec_file>.pid file.
-                self.save_exec_file_pid(child_pid, chroot_exec_file)?;
-                unsafe { libc::exit(0) }
-            }
+        // This is a wrapper for our child callback to the clone libc function.
+        // We will use clone's arg parameter as the actual callback and call it from here.
+        extern "C" fn child_fn(arg: *mut libc::c_void) -> libc::c_int {
+            let callback_ref: *mut *mut dyn FnOnce()  = unsafe { std::mem::transmute(arg) };
+            let callback = unsafe { Box::from_raw(*callback_ref) };
+
+            callback();
+            return 0;
+        }
+
+        let child_callback = || -> () {
```

Comment:
The `()` is not necessary here.

```suggestion
        let child_callback = || {
```